src/common.c

```diff
@@ -1129,10 +1141,24 @@ long get_system_cpus(void) {
 
     debug(D_SYSTEM, "System has %d processors.", processors);
     return processors;
+
+    #endif /* __APPLE__ */
 }
 
 pid_t pid_max = 32768;
 pid_t get_system_pid_max(void) {
+    #ifdef __APPLE__
+        int32_t tmp_pid_max;
+
+        if (unlikely(GETSYSCTL("kern.maxproc", tmp_pid_max))) {
```

Comment:
`kern.maxproc` is not the maximum pid but maximum number of processes. On my macOS VPS kern.maxproc = 532. In FreeBSD we have `kern.pid_max`. I can't find nothing similar in macOS sysctls.